### ISSUE TITLE
Same network mac/IP address bound to multiple containers

### ISSUE THREAD

**Issue by vlerenc** [docker/docker#22334]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

We have multiple Swarm installations. By chance we noticed individual containers having networking issues in one of these Swarm installations running since months (for what it helps: it has experienced multiple issues and updates, i.e. some crashes, process kills, VM recreations reattaching the same OpenStack volume - so far we never experienced issues except once, but then we had volume issues and recreated a Docker VM completely).

Anyway, when I analyzed the issues today, I saw that 4 mac/IP addresses were used by two containers, which seemed odd to me. Another Docker VM in the Swarm cluster was clean. Yet another Docker VM had one duplicate mac/IP addresses. Example:
```
$ docker network inspect bridge | grep -2 02:42:ac:11:00:66
            "07acfa7e548d5b4f5b7e22ef830be249b953ce4e16a4da60586dcf3be592be7d": {
                "EndpointID": "41119c42de63b643f41495fc0acac128038a4ff946720fa4c201cbab699b2d14",
                "MacAddress": "02:42:ac:11:00:66",
                "IPv4Address": "172.17.0.102/16",
                "IPv6Address": ""
--
            "4610aee90f49476f86b0a5a9a841a34e31da1a46cbb5a724b80c6449f5a61348": {
                "EndpointID": "1e47617d7202612219fb64766ebd31f5458c04eb3b02bde2880fe394b9cb36cb",
                "MacAddress": "02:42:ac:11:00:66",
                "IPv4Address": "172.17.0.102/16",
                "IPv6Address": ""
```
Container 07acfa7e548d5b4f5b7e22ef830be249b953ce4e16a4da60586dcf3be592be7d is unresponsive via the IP, but works via localhost. The other container works fine via both:
```
$ docker ps --all | grep -E "07acfa7|4610aee"
07acfa7e548d        stephanme/influxdb:0.9.5.1        "/scripts/run.sh"        5 days ago          Up 2 hours          0.0.0.0:33448->8083/tcp, 0.0.0.0:33449->8086/tcp     cf-b12db439-9019-46a3-9720-6cc2c69670ad
4610aee90f49        joyahan/rabbitmq:3.6.1            "/scripts/run.sh "       11 days ago         Up 11 days          0.0.0.0:33312->5672/tcp, 0.0.0.0:33311->15672/tcp    cf-c3884a11-1852-4768-b090-865e41f3a216

$ nc -v localhost 33448
Connection to localhost 33448 port [tcp/*] succeeded!

$ nc -v localhost 33312
Connection to localhost 33312 port [tcp/*] succeeded!

$ nc -v 10.3.1.1 33448
nc: connect to 10.3.1.1 port 33448 (tcp) failed: Connection refused

$ nc -v 10.3.1.1 33312
Connection to 10.3.1.1 33312 port [tcp/*] succeeded!
```

All containers are created in the same way by the same piece of software using default networking.

**Output of `docker version`:**

```
Client:
Version:      1.9.0
API version:  1.21
Go version:   go1.4.3
Git commit:   76d6bc9
Built:        Tue Nov  3 19:20:09 UTC 2015
OS/Arch:      linux/amd64

Server:
Version:      1.9.0
API version:  1.21
Go version:   go1.4.3
Git commit:   76d6bc9
Built:        Tue Nov  3 19:20:09 UTC 2015
OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 197
Images: 135
Server Version: 1.9.0
Storage Driver: aufs
Root Dir: /var/vcap/store/docker/docker/aufs
Backing Filesystem: extfs
Dirs: 529
Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-49-generic
Operating System: Ubuntu 14.04.4 LTS
CPUs: 4
Total Memory: 7.797 GiB
Name: d4f169b1-c70f-487f-a164-fe4beb029f43
ID: 24RY:G2RP:4ZVH:2TKR:RJS6:QOOF:TSDY:H4XC:ENQL:WF7P:M6JJ:2Z4N
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Running on Ubuntu VMs on OpenStack.

**Comment by aboch** [comment]
Likely a duplicate of #18535

**Comment by thaJeztah** [comment]
@vlerenc can you update to docker 1.10 or 1.11? this issue should be fixed in docker 1.10

**Comment by vlerenc** [comment]
Thank you, we will try to run an update of Docker and let you know should this not resolve the issue.
